.github/workflows/lint.yml

```diff
@@ -0,0 +1,31 @@
+name: Lint Workspaces
```

Comment:
Most of our workflows have been done in Azure DevOps. We should probably get this added to those to have everything housed in a single location. @ryanlengel is this something you can advise on?

Reply:
All the pipelines seem to operate on a single package at a time - I thought it might be worthwhile (and faster) to have a single pipeline that does some static code analysis. We can (hopefully) revisit this later as moving things back into ADO will require a bit more work.

Reply:
I'm fine to go with this for now. Let's speak with Ryan later this week and figure out what the general team stance is. Honestly it seems odd to me that we wouldn't lean into using GitHub Actions more, but the general team stance so far has seemed to push hard towards ADO.

Reply:
I'm open to migrating to Github Actions, I'm tracking this at #760 for us to follow up on later.